README.adoc

```diff
@@ -0,0 +1,275 @@
+for instance:
+
+* `ruby-core` as part of their CI system runs their test suite against an unpackaged Ruby
+  tarball which doesn't have a `.git` folders. That means they needed to overwrite the
+  bundler gemspec to not use git. +
+  Actually, not anymore, https://github.com/rubygems/bundler/pull/6985[since git has been removed from bundler's gemspec].
+
+* If you build your application on a bare docker image without git, and you are pointing to
+  a git sourced gem that uses git on its gemspec, you'll get:
+  `No such file or directory - git ls-files (Errno::ENOENT)` warnings all around. +
+  For example, if you use this in your Gemfile: +
+  `gem "foo", git: "https://github.com/has-git-in-gemspec/foo"`
+
+Originally, `git ls-files` inside the default gemspec template was designed so that users
+publishing their first gem wouldn't unintentionally publish artifacts to it. +
+Now that `bundler` won't let you release if you have uncommitted stuff in your working
+repository, the risk is quite lower (if none at all!).
+
+==== Examples [[gemspec-git-examples]]
+
+[source,ruby]
+----
+# bad
+Gem::Specification.new do |spec|
+  spec.files         = `git ls-files`.split('\n')
+  spec.test_files    = `git ls-files -- spec/*`.split('\n')
+end
+
+# good
+Gem::Specification.new do |spec|
+  spec.files         = Dir['lib/**/*', 'LICENSE', 'README.md']
+  spec.test_files    = Dir['spec/**/*']
+end
+
+# bad
+Gem::Specification.new do |spec|
+  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
+    `git ls-files -z`.split('\\x0').reject { |f| f.match(%r{^(test|spec|features)/}) }
+  end
+end
+
+# good
+Gem::Specification.new do |spec|
+  spec.files         = Rake::FileList['**/*'].exclude(*File.read('.gitignore').split)
+end
+
+# bad
+Gem::Specification.new do |spec|
+  spec.files         = `git ls-files`.split('\n')
+  spec.test_files    = `git ls-files -- test/{functional,unit}/*`.split('\n')
+  spec.executables   = `git ls-files -- bin/*`.split('\n').map { |f| File.basename(f) }
+end
+
+# good
+Gem::Specification.new do |spec|
+  spec.files         = Dir.glob('lib/**/*')
+  spec.test_files    = Dir.glob('test/{functional,test}/*')
+  spec.executables   = Dir.glob('bin/*').map { |f| File.basename(f) }
+end
+----
+
+
+=== Relative Require To Lib [[require-relative-to-lib]]
+
+About the `RelativeRequireToLib` cop:
+
+==== Rationale [[require-relative-to-lib-rationale]]
+
+Avoid using `require_relative` with relative path to lib. Use `require` instead.
+
+==== Need [[require-relative-to-lib-need]]
+
+As mentioned earlier, packages in Debian are built in a clean environment. After building them,
+we also run the tests with the `lib/` directory removed to force those tests to use the
+library installed globally via the Debian package. +
+We do so because we want to ensure that the library is loaded from the system location and not
+from the source tree and that it works like it is intended to.
```

Comment:
I think the explanation for this can be improved. What about this:

Debian has a [testing infrastructure](https://ci.debian.net/) that is designed to test packages in their installed form, i.e. closer to how an end used would use it than to how a developer work against it. For this to work, the test suite must load code that's installed system-wide, instead of the code in the source tree. Using `require_relative` from the tests into the `lib` directory makes that impossible, but it also makes the test look less like client code that would use the code in your gem. Therefore, we recommend that test code uses the main library code without `require_relative`.

Reply:
Nicely phrased, added! :+1: